fe/fe-core/src/main/java/com/starrocks/connector/iceberg/CachingIcebergCatalog.java

```diff
@@ -204,6 +205,24 @@ public Map<String, Partition> getPartitions(String dbName, String tableName, lon
         return partitionCache.getUnchecked(key);
     }
 
+    @Override
+    public List<String> listPartitionNames(String dbName, String tableName, ConnectorMetadatRequestContext requestContext,
+                                           ExecutorService executorService) {
+        // optimization for query mv rewrite, we can optionally return null to bypass it.
+        // if we don't have cache right now, which means it probably takes time to load it during query,
+        // so we can do load in background while return null to bypass this synchronous process.
+        if (requestContext.isQueryMVRewrite()) {
+            long snapshotId = requestContext.getSnapshotId();
+            IcebergTableName key = new IcebergTableName(dbName, tableName, snapshotId);
+            Map<String, Partition> cacheValue = partitionCache.getIfPresent(key);
+            if (cacheValue == null) {
+                partitionCache.refresh(key);
```

Comment:
are you sure this is async?

Reply:
it's async when there is value.
and it's sync when there is no value.

Damn, have to fix this.